pkg/agent/agent_test.go

```diff
@@ -768,32 +791,35 @@ func Test_Running_agent(t *testing.T) {
 					return true, nil, fmt.Errorf("unexpected action, expected %T, got %T", del, action)
 				}
 
-				if _, ok := expectedPodsRemovedNames[deleteAction.Name]; !ok {
-					t.Fatalf("Unexpected pod %q removed", deleteAction.Name)
-				}
+				return testPodRemoval(deleteAction.Name)
+			})
 
-				expectedPodRemovedMutex.Lock()
-				defer expectedPodRemovedMutex.Unlock()
-				expectedPodRemoved--
+			verb := "create"
+			fakeClient.PrependReactor(
+				verb,
+				"pods/eviction",
+				func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
```

Comment:
I think it will be more readable this way
```suggestion
			fakeClient.PrependReactor("create", "pods/eviction",
				func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
```